dbms/src/Storages/Transaction/TiDB.cpp

```diff
@@ -155,10 +155,11 @@ Field ColumnInfo::defaultValueToField() const
         auto v = value.convert<String>();
         if (hasBinaryFlag())
         {
-            // For binary column, we have to pad trailing zeros according to the specified type length.
+            // For some binary column(like varchar(20)), we have to pad trailing zeros according to the specified type length.
```

Comment:
Add some comment to explain why we should check flen

Reply:
fix